skbio/io/format/genbank.py

```diff
@@ -562,8 +562,7 @@ def _parse_locus(lines):
                ' +(bp|aa|rc)'
                ' +(.*DNA|.*RNA)?'
                ' +(linear|circular)?'
-               ' +(PRI|ROD|MAM|VRT|INV|PLN|BCT|VRL|PHG|'
-               'SYN|UNA|EST|PAT|STS|GSS|HTG|HTC|ENV|CON)'
+               ' +(?!.*DNA|.*RNA)([A-Z]{3})'
```

Comment:
Yes, if you read comments both `+(.*DNA|.*RNA)?` and `+(linear|circular)?` could be optional. The problem arises with ```LOCUS       SCU49845     5028 bp    DNA                    21-JUN-1999```, which lack division and must raise an exception. Since the two previous line can be empty, DNA is read by division and so no exception is raised. Maybe this is the reason why you specified all the possible divisions in regular expression.